src/socket/udp_socket.cr

```diff
@@ -89,8 +88,8 @@ class UDPSocket < IPSocket
   # bytes_read, client_addr = server.receive(message)
   # ```
   def receive(message : Bytes) : {Int32, IPAddress}
-    bytes_read, sockaddr, addrlen = system_receive(message)
-    {bytes_read, IPAddress.from(sockaddr, addrlen)}
+    bytes_read, address = system_receive(message)
+    {bytes_read, address.as(IPAddress)}
```

Comment:
Could be shortened to, I think:

```suggestion
    system_receive(message)
```

Reply:
@Sija but `system_receive` returns an `Address` while we want an `IPAddress` here.